src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -846,24 +839,29 @@ private void ReportDiagnostic(ErrorCode errorCode, SyntaxNode syntaxNode, params
             }
         }
 
-        private void InheritNullableStateOfTrackableStruct(TypeSymbol targetType, int targetSlot, int valueSlot, bool isDefaultValue, int slotWatermark)
+        private void InheritNullableStateOfTrackableStruct(TypeSymbol targetType, int targetSlot, int valueSlot, bool isDefaultValue, int stopAtSlot = -1)
         {
             Debug.Assert(targetSlot > 0);
             Debug.Assert(EmptyStructTypeCache.IsTrackableStructType(targetType));
 
+            if (stopAtSlot < 0)
+            {
+                stopAtSlot = targetSlot;
+            }
+
             // https://github.com/dotnet/roslyn/issues/29619 Handle properties not backed by fields.
             // See ModifyMembers_StructPropertyNoBackingField and PropertyCycle_Struct tests.
             foreach (var field in _emptyStructTypeCache.GetStructInstanceFields(targetType))
             {
-                InheritNullableStateOfMember(targetSlot, valueSlot, field, isDefaultValue: isDefaultValue, slotWatermark);
+                InheritNullableStateOfMember(targetSlot, valueSlot, field, isDefaultValue: isDefaultValue, stopAtSlot);
             }
         }
 
-        // 'slotWatermark' is used to avoid inheriting members from inherited members.
-        private void InheritNullableStateOfMember(int targetContainerSlot, int valueContainerSlot, Symbol member, bool isDefaultValue, int slotWatermark)
+        // 'stopAtSlot' is the slot at which to stop copying in case of cycles.
+        private void InheritNullableStateOfMember(int targetContainerSlot, int valueContainerSlot, Symbol member, bool isDefaultValue, int stopAtSlot)
```

Comment:
>stopAtSlot [](start = 139, length = 10)

"stopAtValueSlot"? BTW, is the "stop" portion of the name accurately reflects what is going on? Are we stopping once we get to the slot, or are we skipping the slot?

Reply:
We stop copying for that slot and beneath that slot. We continue to copy other sibling fields.

---
In reply to: [258283173](https://github.com/dotnet/roslyn/pull/33455#discussion_r258283173) [](ancestors = 258283173)

Reply:
> We stop copying for that slot and beneath that slot. We continue to copy other sibling fields.

In this case, I would say we are skipping rather than stopping.

---
In reply to: [258286165](https://github.com/dotnet/roslyn/pull/33455#discussion_r258286165) [](ancestors = 258286165,258283173)